lib/module.js

```diff
@@ -1,27 +1,28 @@
 module.exports = function (moduleOptions) {
-  const extensions = this.options.extensions || []
-  if (!extensions.includes('mdx')) {
+  const { options } = this
+
+  const { extensions } = options
+  if (extensions && !extensions.includes('mdx')) {
     extensions.push('mdx')
   }
 
-  const additionalExtensions = this.options.build.additionalExtensions
-  if (Array.isArray(additionalExtensions)) {
+  const { additionalExtensions } = options.build
+  if (additionalExtensions && !additionalExtensions.includes('mdx')) {
     additionalExtensions.push('mdx')
-  } else if (!additionalExtensions.includes('mdx')) {
-    this.options.build.additionalExtensions = ['mdx']
   }
 
   this.extendBuild((config) => {
-    if (Array.isArray(config.resolve.extensions)) {
+    if (!config.resolve.extensions.includes('.mdx')) {
       config.resolve.extensions.push('.mdx')
-    } else {
-      config.resolve.extensions = ['.mdx']
     }
 
+    const jsxRuleLoaders = config.module.rules.find(r => r.test.test('.jsx')).use
+    const babelLoader = jsxRuleLoaders[jsxRuleLoaders.length - 1]
+
     config.module.rules.push({
       test: /\.mdx$/,
       use: [
-        'babel-loader',
+        babelLoader,
         '@mdx-js/vue-loader'
       ]
     })
```

Comment:
Interesting 😃 . I didn't know it could be done here! Thank you so much! I love Nuxt modules more and more 💚